### PR TITLE
Development

### PR DESCRIPTION
**Pull request by awsumpter**
Review and Merge

### CODE REVIEW

File: Calculator/Calculator.cpp [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-Input from cin; output to cout.
+Input comes from cin through the Token_stream called "ts"; output to cout.
 The grammar for input is :
 Calculation:
     Statement
     Print ';'
     Quit 'q'
 Statement:
-    Declaration
+    Declaration (of variable)
```
Comment (davrdic) [inline Calculator/Calculator.cpp:12]: fix diamond text

File: Calculator/Calculator.cpp [hunk 2]
```diff
@@ -27,18 +27,17 @@ Primary :
     (Expression)
     � Primary
```
Comment (davrdic) [inline Calculator/Calculator.cpp]: fix symbol

File: Calculator/Functions.cpp [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #include <iostream>
 #include <vector>
+#include <Windows.h>
+#include "conio.h"
```
Comment (davrdic) [inline Calculator/Functions.cpp]: remove unnecessary #includes

File: Calculator/Functions.cpp [hunk 4]
```diff
@@ -53,252 +77,419 @@ bool is_declared(string var)
...
+// "double c" is set to the result of function "expression()", Token t is set to
+// the next token in Token_stream 'ts', which is returned by  function "ts.get()"
+// If the kind property of the token is not a ')' an error is thrown
+// "double c" is returned
+
+double parenthesis(Token_stream& myts, vector<Variable>& var_table)
+{
+    double c = expression(myts, var_table);
+    Token t = myts.get();
+    if (t.kind != ')') {
```
Comment (davrdic) [inline Calculator/Functions.cpp]: move all declarations to the top and put a space between declarations and algorithm

File: Calculator/Functions.cpp [hunk 5]
```diff
@@ -53,252 +77,419 @@ bool is_declared(string var)
...
     }
-    ts.putback(t);
-    ts.ignore('(');
-    double e = expression();
+    double e = expression(myts, var_table);
     if (e < 0) {
         error("Square Root of Negative");
     }
-    Token t2 = ts.get();
+    Token t2 = myts.get();
```
Comment (davrdic) [inline Calculator/Functions.cpp]: change myts to something that makes ya feel good

File: Calculator/Functions.cpp [hunk 6]
```diff
@@ -53,252 +77,419 @@ bool is_declared(string var)
...
 
-double primary()
+//------------------------------------------------------------------------------
+// Multiplies each number by the next lowest number starting with the user input until 1 is reached.
+// Example: 5! = 5*4*3*2*1
+// Returns the result.
+
+double calculate_fac(double fact)
 {
+
```
Comment (davrdic) [inline Calculator/Functions.cpp]: remove extra space

File: Calculator/Token_stream.cpp [hunk 7]
```diff
@@ -30,6 +29,14 @@ Token Token_stream::get()
     }
     char ch;
     cin >> ch;
+    if(ch != print)
```
Comment (davrdic) [inline Calculator/Token_stream.cpp]: might can evaluate these in one line...if it evaluates print and fails, will it still evaluate the get. if it doesn't evaluate get and then does unget, we have a problem

Comment (davrdic) [reply Calculator/Token_stream.cpp]: maybe dont worry about it unless you really want to see what happens

File: Calculator/Token_stream.cpp [hunk 8]
```diff
@@ -107,6 +121,7 @@ void Token_stream::ignore(char c)
     char ch = 0;
     while (cin >> ch) {
         if (ch == c) {
+
```
Comment (davrdic) [inline Calculator/Token_stream.cpp]: love it

**Review by davrdic — CHANGES_REQUESTED** [review]
remove calculator.h and fix the other minor things

**Comment by pyronancy** [comment]
hey hey

**Review by pyronancy — APPROVED** [review]
i did it

**Comment by pyronancy** [comment]
is this it?

**Review by davrdic — APPROVED** [review]
is this doing it?

**Pull request merged**